dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java

```diff
@@ -865,4 +874,81 @@ static void assertServerHappy(LibertyServer server) throws Exception {
         assertThat("Server log has errors after starting", server.findStringsInLogs("CWWKF0002E"), empty()); // A bundle could not be found for ...
         assertThat("Server log has errors after starting", server.findStringsInLogs("CWWKE0702E"), empty()); // Could not resolve module ...
     }
+
+    private static Set<String> detectedShortNames = null;
+
+    private static Set<String> getDetectedShortNames() {
+        if (detectedShortNames != null) {
+            return detectedShortNames;
+        }
+
+        detectedShortNames = new HashSet<String>();
+        Bootstrap b;
+        try {
+            b = Bootstrap.getInstance();
+        } catch (Exception e) {
+            Log.info(c, "getDetectedShortNames", "Bootstrap.getInstance() failed. This will likely cause the current repeat to skip " + e.toString());
+            detectedShortNames = null; //if bootstrap is broken we've got bigger problems, but we can avoid caching an exception
+            throw new RuntimeException(e);
+        }
+
+        String installRoot = b.getValue("libertyInstallPath");
+        String libPath = installRoot + "/lib/features/";
+
+        PathMatcher matcher = FileSystems.getDefault().getPathMatcher("glob:*.mf");
+
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(libPath))) {
+            for (Path path : stream) {
+                if (!Files.isDirectory(path) && matcher.matches(path.getFileName())) {
+                    Files.lines(path)
+                                    .filter(line -> line.startsWith("IBM-ShortName: "))
+                                    .map(line -> line.replaceAll("IBM-ShortName: ", ""))
+                                    .forEach(detectedShortNames::add);
+                }
+            }
+        } catch (IOException e) {
```

Comment:
This was new to me, it's very neat.

Docs suggest you can pass in a glob pattern to `newDirectoryStream`, which seems like it would be nice for conciseness.

Docs say that the stream returned from `Files.lines` must be closed to ensure the underlying file is closed. Surprisingly, that doesn't seem to happen automatically when you use a terminal operator.

```suggestion
        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(libPath), "*.mf")) {
            for (Path path : stream) {
                if (!Files.isDirectory(path)) {
                    try (Stream<String> lineStream = Files.lines(path)) {
                        lineStream
                                        .filter(line -> line.startsWith("IBM-ShortName: "))
                                        .map(line -> line.replaceAll("IBM-ShortName: ", ""))
                                        .forEach(detectedShortNames::add);
                    }
                }
            }
        } catch (IOException e) {
```